bin/prover-client/src/operators/checkpoint.rs

```diff
@@ -55,6 +59,24 @@ impl CheckpointOperator {
             .ok_or(ProvingTaskError::WitnessNotFound)
     }
 
+    /// Retrieves the [`L2BlockId`] for the given `block_num`
+    pub async fn get_l2id(&self, block_num: u64) -> Result<L2BlockId, ProvingTaskError> {
+        let l2_headers = self
+            .cl_client
+            .get_headers_at_idx(block_num)
+            .await
+            .inspect_err(|_| error!(%block_num, "Failed to fetch l2_headers"))
+            .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
+
+        let cl_stf_id_buf: Buf32 = l2_headers
+            .expect("invalid height")
+            .first()
+            .expect("at least one l2 blockid")
```

Comment:
Should handle this by printing an error and failing out in case the database is corrupted instead of just panicking.